client/src/legacy/AddToCampaignForm.js

```diff
@@ -69,7 +71,11 @@ jQuery.entwine('ss', ($) => {
       const modalSchemaUrl = `${sectionConfig.form.AddToCampaignForm.schemaUrl}/${id}`;
       const title = i18n._t('Admin.ADD_TO_CAMPAIGN', 'Add to campaign');
 
-      ReactDOM.render(
+      let root = this.getReactRoot();
+      if (!root) {
+        root = createRoot(this[0]);
+      }
+      root.render(
```

Comment:
`ReactDOM.render` was deprecated in react 17 and removed in react 18. It would still _technically_ work but it would result in react behaving like it was still v17. See [this blog post](https://reactjs.org/link/switch-to-createroot) for more details.
See also `_clearModal()` below for the replacement of the old `ReactDOM.unmountComponentAtNode()` method.